### DISCUSSION TITLE
Attributes with nested data structures

### DISCUSSION THREAD

**Discussion by arielvalentin** [open-telemetry/opentelemetry-specification#3624] [Q&A] [answered]
We have a [use case](https://github.com/open-telemetry/opentelemetry-ruby-contrib/pull/557) where we want to capture complex JSON objects in span event attributes. The solution proposed is to serialize the value into a JSON string, but that can be challenging to use effectively in some back end systems.

In the example I liked above we are trying to represent validation errors in span events https://graphql.org/learn/validation/: 

```json
{
  "errors": [
    {
      "message": "Field \"name\" must not have a selection since type \"String!\" has no subfields.",
      "locations": [
        {
          "line": 4,
          "column": 10
        }
      ]
    }
  ]
}
```

Looking at the object I can see some direct mappings to OTel Trace SemConv attributes, while others may be a little more ambiguous:

```json
{
  "events": [
    {
      "name": "Field \"name\" must not have a selection since type \"String!\" has no subfields.",
      "attributes": {
        "graphql.validation.errors": [
          {
            "code.lineno": 4,
            "code.column": 10
          }
        ]
      }
    }
  ]
}
```

All that being said, what is the best way for us to represent these validation errors? 

Should we include them in our instrumentation at all? 

What are other language SIGs doing for these use cases?

cc: @swalkinshaw @open-telemetry/ruby-contrib-approvers @open-telemetry/specs-semconv-approvers

**Comment by SonjaChevre** [comment, marked as answer]
hi @arielvalentin ! 

I opened an issue to discuss the GraphQL semantic convention: https://github.com/open-telemetry/semantic-conventions/issues/182

would be great to join forces, feel free to add your thoughts on the proposal.

**Comment by adriangb** [comment]
A related feature request: https://github.com/open-telemetry/opentelemetry-specification/discussions/3731
